docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile

```diff
@@ -192,3 +192,50 @@ function hs_field_helpers_eck_entity_type_insert(EntityInterface $entity) {
     "edit any $eck_type entities",
   ]);
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function su_humsci_profile_preprocess_menu(&$variables) {
+  if ($variables['menu_name'] != 'shortcut_menu') {
+    return;
+  }
+
+  $current_user = \Drupal::currentUser();
+  _su_humci_profile_clean_shortcut_links($variables['items'], $current_user);
+}
+
+/**
+ * Recursively remove links the current user has no access to.
+ *
+ * @param array $links
+ *   Menu links.
+ * @param \Drupal\Core\Session\AccountInterface $current_user
+ *   Current user object.
+ */
+function _su_humci_profile_clean_shortcut_links(array &$links, AccountInterface $current_user) {
+  foreach ($links as $link_id => $link_item) {
+
+    // This user doesn't have permission for this url. Remove the link.
+    if (!$link_item['url']->access($current_user)) {
+      unset($links[$link_id]);
+      continue;
+    }
+
+    // User has access to the parent menu link, but check all the children.
+    if (!empty($links[$link_id]['below'])) {
+      _su_humci_profile_clean_shortcut_links($links[$link_id]['below'], $current_user);
+    }
+  }
+}
+
+/**
+ * Implements hook_page_attachments().
+ */
+function su_humsci_profile_page_attachments(array &$attachments){
+  // Conditionally attach an asset to the page.
+  if (!\Drupal::currentUser()
```

Comment:
Do we want to also add a check here so that we avoid attaching this css for anonymous users?

Reply:
good thinkin. :+1:

Reply:
instead of checking for anonymous, i checked if user has permission to view the toolbar. essentially it results in the same, but would also avoid adding the css when we go about a private page functionality later.